### PR TITLE
[#2839] improvment(CI): add log of testcontainers

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

Separate TestContainers logs into a different log file.

### Why are the changes needed?

Fix: #2839 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

by hand

### CODE REVIEW

File: catalogs/catalog-hadoop/src/test/resources/log4j2.properties [hunk 1]
```diff
@@ -23,11 +23,37 @@ appender.file.type = File
...
 rootLogger.appenderRef.file.ref = fileLogger
+
+# File appender configuration for testcontainers
+appender.testcontainersFile.type = File
+appender.testcontainersFile.name = testcontainersLogger
+appender.testcontainersFile.fileName = build/testcontainers.log
+appender.testcontainersFile.layout.type = PatternLayout
+appender.testcontainersFile.layout.pattern = %d{yyyy-MM-dd HH:mm:ss.SSS} [%t] %-5p %c - %m%n
+
+# Logger for testcontainers
```
Comment (mchades) [inline catalogs/catalog-hadoop/src/test/resources/log4j2.properties:42]: Reference document: https://java.testcontainers.org/supported_docker_environment/logging_config/

**Comment by jerryshao** [comment]
Will it have many logs?

**Comment by mchades** [comment]
> Will it have many logs?

I uploaded the log file, you can download it and take a look. I think it is fine

**Comment by jerryshao** [comment]
@yuqi1129 can you please take a review, I'm fine with it.

File: integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/AbstractIT.java [hunk 2]
```diff
@@ -228,7 +228,7 @@ public static void startIntegrationTest() throws Exception {
 
   @AfterAll
   public static void stopIntegrationTest() throws IOException, InterruptedException {
-    if (testMode != null && testMode.equals(ITUtils.EMBEDDED_TEST_MODE)) {
+    if (testMode != null && testMode.equals(ITUtils.EMBEDDED_TEST_MODE) && miniGravitino != null) {
```
Comment (yuqi1129) [inline integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/AbstractIT.java:231]: is it a bug that you fixed by the way?

Comment (mchades) [reply integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/AbstractIT.java:231]: yes

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**